lib/test.js

```diff
@@ -104,51 +112,52 @@ Test.prototype.plan = function (count) {
 };
 
 Test.prototype.run = function () {
-	this.promise = {};
+	var self = this;
+
+	this.promise = Promise.pending();
 
 	// TODO(vdemedes): refactor this to avoid storing the promise
-	return new Promise(function (resolve, reject) {
-		this.promise.resolve = resolve;
-		this.promise.reject = reject;
+	if (!this.fn) {
+		return this.exit();
```

Comment:
I prefer not an empty `return` if it actually doesn't return anything. It's more explicit and has a better intent.

```js
this.exit();
return;
```